lib/collection/array.js

```diff
@@ -158,33 +157,31 @@ export class ArrayColl extends KeyValueCollection {
   }
 
   removeKey(i) {
-    var item = this._array[i];
-    if (item == undefined) {
-      return;
+    if (i in this._array) {
+      this._notifyRemoved(this._array.splice(i, 1));
     }
-    delete this._array[i];
```

Comment:
I think we still need this line, no? Otherwise this function doesn't actually remove anything.

Reply:
Please also add a test for this.